src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/AdministrationController.php

```diff
@@ -170,8 +163,9 @@ protected function processForm(Request $request, FormHandlerInterface $formHandl
             $data = $form->getData();
             try {
                 $formHandler->save($data);
-            } catch (DataProviderException $e) {
-                $this->flashErrors($this->getErrorMessages($e->getInvalidConfigurationDataErrors()));
+            } catch (FormDataProviderException $e) {
+                $errorMessageFactory = $this->get('prestashop.form.error_message.configuration_error_factory');
```

Comment:
I think for new service call, we should fit with https://github.com/PrestaShop/ADR/blob/master/0019-FQCN-Autowiring.md. You probably should create FQCN version of prestashop.form.error_message.configuration_error_factory and use it here.